compio-fs/src/metadata/mod.rs

```diff
@@ -0,0 +1,9 @@
+cfg_if::cfg_if! {
+    if #[cfg(windows)] {
+        mod windows;
+        pub use windows::*;
```

Comment:
Don't expose two different types on two target. Wrap it into one for a more consistent API.

See [std implementation](https://doc.rust-lang.org/src/std/fs.rs.html#112).